tests/nested/classic/hotplug/task.yaml

```diff
@@ -0,0 +1,297 @@
+summary: Create ubuntu classic image, install snapd and test hotplug feature
+
+prepare: |
+    #shellcheck source=tests/lib/nested.sh
+    . "$TESTSLIB/nested.sh"
+    create_nested_classic_vm
+
+    copy_remote "${GOHOME}"/snapd_*.deb
+    execute_remote "sudo apt update"
+    execute_remote "sudo apt install -y ./snapd_*.deb"
+    #shellcheck disable=SC2016
+    execute_remote 'DEBIAN_FRONTEND=noninteractive sudo -E apt install -y linux-image-extra-$(uname -r) || DEBIAN_FRONTEND=noninteractive sudo -E apt install -y linux-modules-extra-$(uname -r)'
+
+    snap pack "$TESTSLIB"/snaps/serial-port-hotplug
+    copy_remote serial-port-hotplug_1.0_all.snap
+    # add test user to dialout group required for ttyUSB* devices
+    execute_remote "sudo usermod -a -G dialout user1"
+
+    execute_remote "sudo snap install --devmode jq"
+
+restore: |
+    #shellcheck source=tests/lib/nested.sh
+    . "$TESTSLIB/nested.sh"
+    destroy_nested_vm
+    rm -rf /tmp/serialport{0,1}
+
+debug: |
+    #shellcheck source=tests/lib/nested.sh
+    . "$TESTSLIB/nested.sh"
+
+    execute_remote "snap interfaces" || true
+    execute_remote "snap list" || true
+    execute_remote "dmesg" || true
+    execute_remote 'sudo jq -r ".data[\"hotplug-slots\"] /var/lib/snapd/state.json' || true
+
+execute: |
+    hotplug_add_dev1() {
+        add_tty_chardev my-chardev /tmp/serialport1
+        add_usb_serial_device my-usb-serial my-chardev 1234
+    }
+
+    hotplug_del_dev1() {
+        del_device my-usb-serial
+        remove_chardev my-chardev
+    }
+
+    hotplug_add_dev2() {
+        add_tty_chardev my-chardev2 /tmp/serialport2
+        add_usb_serial_device my-usb-serial2 my-chardev2 5678
+    }
+
+    hotplug_del_dev2() {
+        del_device my-usb-serial2
+        remove_chardev my-chardev2
+    }
+
+    # Check that given slot is not present in 'snap interfaces' output
+    # (note, it can still be present in the state with hotplug-gone=true)
+    check_slot_not_present() {
+        SLOT_NAME="$1"
+        for _ in $(seq 10); do
+            if ! execute_remote "snap interfaces" | MATCH ":$SLOT_NAME"; then
```

Comment:
can we use "connections" now instead?

Reply:
Yep! This uncovered a bug in integration between connections & hotplug - HotplugGone is not honored! Bugifx coming in separate PR.

Reply:
Changed to use snap connections, but it needs #6590 to land first.